packages/block-editor/src/components/block-media-update-progress/test/index.native.js

```diff
@@ -0,0 +1,501 @@
+		mime: 'image/jpeg',
+		type: 'image',
+		url: '',
+	},
+	{
+		alt: '',
+		caption: '',
+		id: '3',
+		link: '',
+		mime: 'image/jpeg',
+		type: 'image',
+		url: '',
+	},
+];
+
+describe( 'BlockMediaUpdateProgress component', () => {
+	it( 'renders without crashing', () => {
+		const wrapper = shallow(
+			<BlockMediaUpdateProgress renderContent={ () => {} } />
+		);
+		expect( wrapper ).toBeTruthy();
+	} );
+
+	it( 'upload: listens media upload progress for local file', () => {
+		const progress = 10;
+		const payload = {
+			state: MEDIA_UPLOAD_STATE_UPLOADING,
+			mediaId: MEDIA_ID_LOCAL,
+			progress,
+		};
+
+		const onUpdateMediaUploadProgress = jest.fn();
+
+		const wrapper = shallow(
+			<BlockMediaUpdateProgress
+				onUpdateMediaUploadProgress={ onUpdateMediaUploadProgress }
+				mediaFiles={ localMediaFiles }
+				renderContent={ () => {} }
+			/>
+		);
+
+		sendMediaUpload( payload );
+
+		expect( wrapper.instance().state.progress ).toEqual( progress );
+		expect( wrapper.instance().state.isUploadInProgress ).toEqual( true );
+		expect( wrapper.instance().state.isUploadFailed ).toEqual( false );
+		expect( onUpdateMediaUploadProgress ).toHaveBeenCalledTimes( 1 );
+		expect( onUpdateMediaUploadProgress ).toHaveBeenCalledWith( payload );
+	} );
+
+	// UPLOAD tests
+	it( 'upload does not get affected by unrelated media uploads', () => {
+		const payload = {
+			state: MEDIA_UPLOAD_STATE_UPLOADING,
+			mediaId: 432, // id not belonging to assigned mediaFiles collection in test
+			progress: 20,
+		};
+		const onUpdateMediaUploadProgress = jest.fn();
+		const wrapper = shallow(
+			<BlockMediaUpdateProgress
+				onUpdateMediaUploadProgress={ onUpdateMediaUploadProgress }
+				mediaFiles={ localMediaFiles }
+				renderContent={ () => {} }
+			/>
+		);
+
+		sendMediaUpload( payload );
+
+		expect( wrapper.instance().state.progress ).toEqual( 0 );
+		expect( onUpdateMediaUploadProgress ).toHaveBeenCalledTimes( 0 );
+	} );
+
+	it( 'upload: listens media upload success', () => {
+		const progress = 10;
+		const payloadSuccess = {
+			state: MEDIA_UPLOAD_STATE_SUCCEEDED,
+			mediaId: MEDIA_ID_LOCAL,
+		};
+		const payloadUploading = {
+			state: MEDIA_UPLOAD_STATE_UPLOADING,
+			mediaId: MEDIA_ID_LOCAL,
+			progress,
+		};
+
+		const onFinishMediaUploadWithSuccess = jest.fn();
+
+		const wrapper = shallow(
+			<BlockMediaUpdateProgress
+				onFinishMediaUploadWithSuccess={
+					onFinishMediaUploadWithSuccess
+				}
+				mediaFiles={ localMediaFiles }
+				renderContent={ () => {} }
+			/>
+		);
+
+		sendMediaUpload( payloadUploading );
+
+		expect( wrapper.instance().state.progress ).toEqual( progress );
+
+		sendMediaUpload( payloadSuccess );
+
+		expect( wrapper.instance().state.isUploadInProgress ).toEqual( false );
+		expect( onFinishMediaUploadWithSuccess ).toHaveBeenCalledTimes( 1 );
+		expect( onFinishMediaUploadWithSuccess ).toHaveBeenCalledWith(
+			payloadSuccess
+		);
+	} );
+
+	it( 'upload: listens media upload fail', () => {
```

Comment:
same comment as above. Could we make this name a bit more descriptive? Also same thought process about multiple `expect`/ assertions as seen below.